src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -265,16 +257,84 @@ static ImmutableArray<string> AttachComplexTypes(ImmutableArray<string> receiver
             }
         }
 
-        private static void GetExtensionMethodItemsForSymbolsFromDifferentCompilation(
+        private static SerializableImportCompletionItem CreateItem(IMethodSymbol methodSymbol, Dictionary<INamespaceSymbol, string> stringCache)
+            => new SerializableImportCompletionItem(
+                SymbolKey.CreateString(methodSymbol),
+                methodSymbol.Name,
+                methodSymbol.Arity,
+                methodSymbol.GetGlyph(),
+                GetFullyQualifiedNamespaceName(methodSymbol.ContainingNamespace, stringCache));
+
+        private static ImmutableArray<IMethodSymbol> GetExtensionMethodSymbolsFromPeReference(
+            PortableExecutableReference peReference,
+            SemanticModel semanticModel,
+            ITypeSymbol receiverTypeSymbol,
+            ImmutableArray<string> receiverTypeNames,
+            SymbolTreeInfo index,
+            int position,
+            ISet<string> namespaceFilter,
+            ConcurrentDictionary<ITypeSymbol, bool> checkedReceiverTypes,
+            StatisticCounter counter,
+            CancellationToken cancellationToken)
```

Comment:
i would reocmmend in a future pr breaking this into a class.  I think 10 params is def the trigger point that indicates things have gone too far.